samples/solution-counters.js

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+const { Firestore, FieldValue } = require('@google-cloud/firestore');
+
+async function main() {
+  // [START create_counter]
+  async function createCounter(docRef, numShards) {
+    const colRef = docRef.collection("shards");
+    for (let count = 0; count < numShards; count++) {
+      const newDoc = colRef.doc(count.toString());
+      await newDoc.set({ count: 0 });
+    }
+  }
+  // [END create_counter]
+
+  // [START increment_counter]
+  async function incrementCounter(docRef, numShards) {
+    const shardId = Math.floor(Math.random() * numShards);
+    const shardRef = docRef.collection("shards").doc(shardId.toString());
+    return shardRef.update({ count: FieldValue.increment(1) });
```

Comment:
If you replaced this with 

```
return shardRef.set({ count: FieldValue.increment(1) }, {merge: true});
```

you would not have to create the counters and initialize them with zero.